test/test.js

```diff
@@ -3,17 +3,35 @@ import 'babel-polyfill';
 
 import assert from 'assert';
 import { exec } from 'mz/child_process';
-import { exists, readFile, writeFile } from 'mz/fs';
+import { exists, readFile, writeFile, mkdtemp, mkdir } from 'mz/fs';
+import { join, sep, normalize } from 'path';
+
 
 import getFilesUnderPath from '../src/util/getFilesUnderPath';
 
 let originalCwd = process.cwd();
 
+async function mkdTempSafe (prefix) {
+  let parts = normalize(prefix).split(sep);
+  // let last = parts.pop();
+  let head = '';
+  for (let part of parts) {
+    try {
+      head = join(head, part);
+      await mkdir(head);
+    } catch (e) {
+      if (e.code === 'EEXIST') continue;
+      throw e;
+    }
+  }
+  return await mkdtemp(prefix);
```

Comment:
Thanks, I hadn't seen `mkdtemp`. Does this fix anything for Windows, or is it just a more robust alternative to my manual random directory thing?